xbmc/cores/VideoPlayer/Edl.cpp

```diff
@@ -685,9 +685,19 @@ bool CEdl::AddCut(Cut& cut)
     int autowind = g_advancedSettings.m_iEdlCommBreakAutowind * 1000; // seconds -> ms
 
     if (cut.start > 0) // Only autowait if not at the start.
-     cut.start += autowait;
-    if (cut.end > cut.start + autowind) // Only autowind if it won't go back past the start (should never happen).
-     cut.end -= autowind;
+    {
+      /* get the cut length so we don't start skipping after the end */
+      int cutLength = cut.end - cut.start;
+      /* add the lesser of the cut length or the autowait to the start */
+      cut.start += autowait > cutLength ? cutLength : autowait;
+    }
+    if (cut.end > cut.start) // Only autowind if there is any cut time remaining.
+    {
+      /* get the remaining cut length so we don't rewind to before the start */
+      int cutLength = cut.end - cut.start;
+      /* subtract the lesser of the cut length or the autowind from the end */
+      cut.end -= autowind > cutLength ? cutLength : autowind;
+    }
```

Comment:
I'm hesitant to comment because of the don't comment rule, but I believe this is meaningful.
Unless my logic is flawed, this code would not result in a negative cut because the cutLength is recalculated as the remaining cutLength after the autowait is applied to the cut.start.
In the scenario outlined, cut.start would end up equal to cut.end, resulting in a 0 length cut.
The previous code would have handled this scenario by ignoring the autowind. 
I feel the better solution is to disregard the cut if the cutLengh < (autowait + autowind).

The scenario in which autowait > (cut.end - cut.start) is one that I also though would become more likely if the change to increase the maximum autowait and autowind values was accepted. I don't believe that scenario was handled by the original code.

Thanks for taking the time to look at this change even though I was not able to complete the checklist.